src/Analysis/GWASResultsV2/Utils/gwasWorkflowApi.js

```diff
@@ -0,0 +1,36 @@
+import { headers } from '../../../configs';
+import { gwasWorkflowPath } from '../../../localconf';
+import { getPresignedUrl } from '../../AnalysisJob';
+
+export const fetchPresignedUrlForWorkflowArtifact = async (
+  workflowName,
+  workflowUid,
+  artifactName,
+) => {
+  // query argo-wrapper endpoint to get the list of artifacts produced by the workflow:
+  const endPoint = `${gwasWorkflowPath}status/${workflowName}?uid=${workflowUid}`;
+  const response = await fetch(endPoint, { headers })
+    .then((res) => res.json())
+    .then((data) => data);
+  if (!response) {
+    throw new Error('Error while querying workflow artifacts');
+  }
+  if (!response.outputs.parameters) {
+    throw new Error('Found no artifacts for workflow');
+  }
+  // filter the list to find the artifact matching the given name:
+  const results = response.outputs.parameters.filter((entry) => entry.name === artifactName);
+  if (results.length !== 1) {
+    throw new Error(`Expected 1 artifact with name ${artifactName}, found: ${results.length}`);
+  }
+  // return a pre-signed "download ready" URL to the artifact:
+  return getPresignedUrl(JSON.parse(results[0].value).did, 'download');
+};
+
+// TODO - reuse from ../../GWASApp/Utils/cohortMiddlewareApi.js instead?
```

Comment:
Comment: Can we remove this comment and instead create a ticket for this? My preference would be to keep the code base free from comments like this, they can get out of hand as the app grows.

Reply:
Per our discussion: this code duplication is probably OK, comment can be deleted for now. A new should be created: 

This ticket is: 
* Goal: Create a util file or folder that will house this object that needs to be shared across the GWASApp and the GWASResultsV2 app

Reply:
TODO removed.

Reply:
Ticket created here: https://ctds-planx.atlassian.net/browse/VADC-509